AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOperation.swift

```diff
@@ -157,7 +159,23 @@ final class InitialSyncOperation: AsynchronousOperation {
         let items = syncQueryResult.items
         recordsReceived += UInt(items.count)
 
+        if lastSyncTime != nil {
+            modelSyncedPayload?.isFullSync = false
+            modelSyncedPayload?.isDeltaSync = true
+        } else {
+            modelSyncedPayload?.isFullSync = true
+            modelSyncedPayload?.isDeltaSync = false
+        }
+
         for item in items {
+            let itemMetadata = item.syncMetadata
+            if itemMetadata.deleted {
+                modelSyncedPayload?.deleteCount += 1
+            } else if itemMetadata.version == 1 {
+                modelSyncedPayload?.createCount += 1
```

Comment:
This goes back to a previous discussion: What does "create" mean in this context: that the incoming instance has never been seen *locally*, or that the incoming instance has been created but not updated *on the service*. In other words, in the following case:

- Create Instance1 on console; version == 1
- Update Instance1 on console; version == 2
- Launch app, synchronize
- Receive Instance1; version == 2

Should that be counted as a `create` or an `update`?

Reply:
I also synced up with Rafael on this:
From the perspective of local store, the record is new (no matter is the first time to save it or to delete or save it again)